### ISSUE TITLE
Does running simple_test.py require a lot of cpu memory?

### ISSUE THREAD

**Issue by fei88988** [facebookresearch/pifuhd#119]
When I tried to run the test, the process was always killed by the system because it exceeded the memory range. The total memory of my CPU was 16G. Before the test process was killed, the CPU utilization of the test process reached 500%.

**Comment by fei88988** [comment]
Can someone answer it, I will be grateful to your answer !

**Comment by manncodes** [comment]
Yes it does. I don't know much about Mac CPU utilization, but for Windows 8G CPU utilization will be 100% for time depending on the resolution parameter.
256x256 -> 50 sec
512x512 -> 3-4 min
1024x1024 -> 7 min

to improve memory usage, do following changes in recon.py:

-add following line after imports:
```python
os.environ["KMP_DUPLICATE_LIB_OK"]="TRUE"
```
-after importing torch add line:
```python
torch.cuda.empty_cache()
```
-for switching to CPU change line 145 to:
```python
cuda = torch.device('cpu')
```
hope it helps.

**Comment by GabbyYam** [comment]
exactly the same problem, did you get a solution now?
